django/db/backends/postgresql/schema.py

```diff
@@ -274,6 +274,25 @@ def _alter_column_collation_sql(
             [],
         )
 
+    def _need_to_add_new_index(self, old_field, new_field):
+        if not (old_field.db_index or old_field.unique) and (
+            new_field.db_index or new_field.unique
+        ):
+            return True
+
+    def _need_to_recreate_deleted_index(self, old_field, new_field, old_type, new_type):
+        if (
+            old_field.db_index
+            and not old_field.unique
+            and (
+                not new_field.db_index
+                or (new_field.unique and not new_field.primary_key)
+            )
+        ) or (
+            old_type != new_type and not new_field.unique and not old_field.primary_key
+        ):
```

Comment:
Can you point me to where all indexes are dropped due to type change? As far as I'm aware, this should not be required to check this directly.

Reply:
@felixxm Here's where the _like index is dropped when changing certain types: https://github.com/django/django/blob/main/django/db/backends/postgresql/schema.py#L158.

I could move that check out into its own method, and we could call the same method when we're seeing if we need to recreate the index.

Reply:
I updated the check to match the check where the _like index is dropped.

Note: there seems to be an error on a couple of the job runners:
```
Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
```